src/coin/skycoin/models/wallet.go

```diff
@@ -0,0 +1,273 @@
+	return &SkycoinWalletIterator{wallets: wallets, current: -1}
+}
+
+type WalletService struct { //Implements WalletStorage and WalletSet interfaces
+}
+
+func (wltSrv *WalletService) ListWallets() core.WalletIterator {
+	c := util.NewClient()
+	wlts, err := c.Wallets()
+	if err != nil {
+		return nil
+	}
+	wallets := make([]Wallet, 0)
+	for _, wlt := range wlts {
+		wallets = append(wallets, walletResponseToWallet(wlt))
+	}
+	return NewSkycoinWalletIterator(wallets)
+}
+
+func (wltSrv *WalletService) CreateWallet(label string, seed string, IsEncrypted bool, pwd core.PasswordReader, scanAddressesN int) (core.Wallet, error) {
+	if IsEncrypted {
+		password, _ := pwd("Enter your password")
+		return wltSrv.createEncryptedWallet(seed, label, password, scanAddressesN)
+	} else {
+		return wltSrv.createUnencryptedWallet(seed, label, scanAddressesN)
+	}
+}
+func (wltSrv *WalletService) createEncryptedWallet(seed, label, password string, scanN int) (core.Wallet, error) {
+	c := util.NewClient()
+	wltR, err := c.CreateEncryptedWallet(seed, label, password, scanN)
+	if err != nil {
+		return nil, err
+	}
+
+	wlt := walletResponseToWallet(*wltR)
+	return &wlt, nil
+
+}
+
+func (wltSrv *WalletService) createUnencryptedWallet(seed, label string, scanN int) (core.Wallet, error) {
+	c := util.NewClient()
+	wltR, err := c.CreateUnencryptedWallet(seed, label, scanN)
+	if err != nil {
+		return nil, err
+	}
+	wlt := walletResponseToWallet(*wltR)
+	return &wlt, nil
+}
+
+func (wltSrv *WalletService) Encrypt(walletName string, pwd core.PasswordReader) {
+	c := util.NewClient()
+	password, _ := pwd("Insert password")
+	_, err := c.EncryptWallet(walletName, password)
+	if err != nil {
+		return
+	}
+}
+
+func (wltSrv *WalletService) Decrypt(walletName string, pwd core.PasswordReader) {
+	c := util.NewClient()
+	password, _ := pwd("Insert password")
+	_, err := c.DecryptWallet(walletName, password)
+	if err != nil {
+		return
+	}
+}
+
+func (wltSrv *WalletService) IsEncrypted(walletName string) (bool, error) {
+	c := util.NewClient()
+	wlt, err := c.Wallet(walletName)
+	if err != nil {
+		return false, err
+	}
+	return wlt.Meta.Encrypted, nil
+}
+func (wltSrv *WalletService) GetWallet(id string) core.Wallet {
+	c := util.NewClient()
+	wltR, err := c.Wallet(id)
+	if err != nil {
+		return nil
+	}
+	return walletResponseToWallet(*wltR)
+}
+
+type WalletNode struct { //Implements WallentEnv interface
+	wltService *WalletService
+}
+
+func (wltEnv *WalletNode) GetStorage() core.WalletStorage {
+	if wltEnv.wltService == nil {
+		wltEnv.wltService = new(WalletService)
+	}
+	return wltEnv.wltService
+}
+
+func (wltEnv *WalletNode) GetWalletSet() core.WalletSet {
+	if wltEnv.wltService == nil {
+		wltEnv.wltService = new(WalletService)
+	}
+	return wltEnv.wltService
+}
+
+type SeedService struct{} //Implements SeedGenerator interface
+
+func (seedService *SeedService) GenerateMnemonic(entropy int) (string, error) {
+	c := util.NewClient()
+	seed, err := c.NewSeed(entropy)
```

Comment:
This should be done through `core.SeedGenerator` interface . Most nodes run with `INSECURE_WALLET_SEED` disabled. Hence this call will always lead to (HTTP `404 Not Found`) error. Hence there should be two alternatives:

- `SkycoinRemoteSeedGenerator` invoking client's `NewSeed()` via REST API
- `SkycoinDeafultSeedGenerator` executing seed generation code directly